### DISCUSSION TITLE
How to disable pages-build-deployment workflow?

### DISCUSSION THREAD

**Discussion by aadimator** [quarto-dev/quarto-cli#5810] [Q&A] [answered]
### Description

I want to disable/delete the GitHub Action workflow that gets generated after running the `quarto publish gh-pages` command. I have a separate GitHub Action workflow that I want to use, and I don't use `GitHub Pages` for publishing, but `Cloudflare's CDN`. So I have setup the logic there.

The problem is that whenever I push updates to the GH repo, it automatically triggers 2 GH Actions, one that I want to execute and the other `pages-build-deployment` that I want to disable. 

I've tried deleting or disabling the workflow, but there's no option to do that there, unlike normal GH Actions, where I can disable or remove a workflow. 
![image](https://github.com/quarto-dev/quarto-cli/assets/10383542/8b543f0b-e61d-473c-b3e7-66af75ca5a14)

I've tried using the `Delete workflow run` option as well. Once all the runs are deleted, the `pages-build-deployment` workflow vanishes from the sidebar, but upon new code push, it gets generated again. 
![image](https://github.com/quarto-dev/quarto-cli/assets/10383542/851f78e5-8f91-4741-9ae7-b39a0c401409)

Is there anyway to permanently disable/delete this workflow? I'd really appreciate your help. Thanks

**Comment by mcanouil** [comment]
Why are you running `quarto publish` in the first place here?
Simply commit your changes and push.

Note that this workflow is GitHub and has nothing to do with Quarto.
Remove the GitHub Pages directory/branch to avoid the GitHub Pages being deployed.
Or disable GitHub Pagesw

**Reply by aadimator** [reply]
I ran that command only once, because it was said so in the official doc: 
![image](https://github.com/quarto-dev/quarto-cli/assets/10383542/9ad457c5-0a78-483e-8b7a-e47118487510)

Didn't run that command since then.

I had manually disabled GitHub pages and removed the GH Pages environment as well, but they both get created automatically once the workflow runs. I haven't been able to find any way yet to disable the workflow. Thought that you guys might have some clue.

**Reply by aadimator** [reply]
Oh, I think I know what's the problem here. This might be getting published because of this section in my GH Action yaml file.

```yaml
      - name: Render and Publish
        uses: quarto-dev/quarto-actions/publish@v2
        with:
          target: gh-pages
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

I think I need to use the following script instead:

```yaml
- name: Render Quarto Project
  uses: quarto-dev/quarto-actions/render@v2
```

Can I publish the result of this render action to some other branch, like `gh-pages` or `public` etc? Is it possible with the provided render action or do I have to do that using some other way?

**Reply by mcanouil** [reply, marked as answer]
`gh-pages` is a special branch for GitHub Pages.
If you don't want GitHub Pages, don't use that branch.

Add whatever steps you want after the Quarto render step.

`quarto publish` is a simple wrapper of `quarto render` and common tasks.

See <https://docs.github.com/en/actions>.

**Reply by aadimator** [reply]
Thanks a bunch for all the help. Much appreciated.

**Reply by burgerga** [reply]
A little late, but If you want a convenient way to use a different branch (e.g. `cf-pages` for Cloudflare Pages) have a look at https://github.com/quarto-dev/quarto-cli/discussions/1776#discussioncomment-11893175.

**Comment by fullstackzach** [comment]
Landed here via Google. I wanted to disable github pages completely. 

For those who are looking to do the same, you can choose "None" at the bottom of the branch dropdown to disable it. This is under repo settings General -> Pages

**Reply by jouellnyc** [reply]
Thank you @fullstackzach - this was non obvious to me.

**Reply by jorenbroekema** [reply]
Hahah wow that's so dumb. A simple remove button would be a lot more user-friendly. There's no way you would intuitively know to scroll down the list of branches to a magic "None" to disable this.
